File: Singleton/Singleton.cpp
/*
 *  Singleton.cpp
 *  DesignPatterns
 *
 *  Created by John Therrell on 1/30/11.
 *  Copyright 2011 __MyCompanyName__. All rights reserved.
 *
 */

#include "Singleton.h"

Singleton* Singleton::s_instance = 0;

Singleton::Singleton() {
    foo = 0;
}

Singleton* Singleton::getInstance () {
    if (!s_instance) {
        s_instance = new Singleton;
    }
    return Singleton::s_instance;
}

void Singleton::addToFoo(int num) {
    foo += num;
}

int Singleton::getFoo() {
    return foo;
}

File: Java/src/designpatterns/Singleton.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package designpatterns;

/**
 *
 * @author jtherrell
 */
public class Singleton
{
	private static Singleton s_instance;
	private int foo;

	private Singleton() {
		foo = 0;
	}

	public static Singleton getInstance() {
		if (s_instance == null) {
			s_instance = new Singleton();
		}
		return s_instance;
	}

	public void addToFoo(int num) {
		foo += num;
	}

	public int getFoo() {
		return foo;
	}
}
